src/tpm2_pytss/types.py

```diff
@@ -541,7 +541,52 @@ def __iter__(self):
         return TPML_Iterator(self)
 
 
-class TPMU_PUBLIC_PARMS(TPM_OBJECT):
+class TPMU_OBJECT(TPM_OBJECT):
```

Comment:
Probably need todo #660 and massage the request and refactor it so we have:
TPM_OBJECT becomes TPM_BASE_OBJECT which is a base class without marshaling support
TPM_OBJECT would extend TPM_BASE_OBJECT and add marshalling support/unmarshalling
TPMU_OBJECT would extend TPM_BASE_OBJECT and add marshalling support/unmarshalling with the extra parameter for the union

Reply:
It doesn't work for TPM2_HANDLE as it isn't a structure, so that we would need to keep the M/U methods for TPM2_HANDLE

Reply:
> It doesn't work for TPM2_HANDLE as it isn't a structure, so that we would need to keep the M/U methods for TPM2_HANDLE

Can't you just detect that the underlying type is a scalar and its size and invoke the correct `Tss2_MU_UINT32_Marshal` and unmarshal routine, where 32 is replaced with the scalar's size?

Reply:
Can't that be smartened up to say if scalar do this else do existing behavior?

Reply:
I think that would just make the code more complicated and unreadable.
Having one separate mixin class for integer based types would remove the complexity, it could be shared between TPM2_HANDLE, the different constants and the (U)INT* stuff in command_parser

Reply:
It's fixed now